src/components/swap/AdvancedSwapDetails.test.tsx

```diff
@@ -0,0 +1,46 @@
+import userEvent from '@testing-library/user-event'
+import {
+  TEST_ALLOWED_SLIPPAGE,
+  TEST_TOKEN_1,
+  TEST_TRADE_EXACT_INPUT,
+  TEST_TRADE_EXACT_OUTPUT,
+  toCurrencyAmount,
+} from 'test-utils/constants'
+import { render, screen } from 'test-utils/render'
+
+import { AdvancedSwapDetails } from './AdvancedSwapDetails'
+
+describe('AdvancedSwapDetails.tsx', () => {
+  it('matches base snapshot', () => {
+    const { asFragment } = render(
+      <AdvancedSwapDetails trade={TEST_TRADE_EXACT_INPUT} allowedSlippage={TEST_ALLOWED_SLIPPAGE} />
+    )
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  it('renders correct copy on mouseover', async () => {
+    render(<AdvancedSwapDetails trade={TEST_TRADE_EXACT_INPUT} allowedSlippage={TEST_ALLOWED_SLIPPAGE} />)
+    userEvent.hover(screen.getByText('Price Impact'))
```

Comment:
Wrap your userEvent actions in `await act(() => XYZ)`.
If you do, I think you can use the synchronous getByText for your assertions :)